backend/src/main/java/com/htecgroup/skynest/service/UserService.java

```diff
@@ -1,31 +1,34 @@
 package com.htecgroup.skynest.service;
 
 import com.htecgroup.skynest.model.dto.UserDto;
+import com.htecgroup.skynest.model.request.UserRegisterRequest;
+import com.htecgroup.skynest.model.response.UserResponse;
+import com.htecgroup.skynest.model.request.UserPasswordResetRequest;
 
 import java.util.List;
 import java.util.UUID;
 
 public interface UserService {
 
-  UserDto registerUser(UserDto userDto);
+  UserResponse registerUser(UserRegisterRequest userRegisterRequest);
```

Comment:
Since this interface and class impl with it is becoming more and more bigger, maybe it is good time to separate things between user manipulation and auth logic. Having that said, all methods that are connected by any means with auth such as (confirmEmail, verifyUser, email sending methods, password change etc) should go in new class.

Reply:
Agreed, we were thinking of doing that anyways. Better to just do it now